src/main/java/io/camunda/testing/assertions/ProcessInstanceAssertions.java

```diff
@@ -0,0 +1,413 @@
+   * process has been completed or terminated.
+   *
+   * @return this {@link ProcessInstanceAssertions}
+   */
+  public ProcessInstanceAssertions isStarted() {
+    final Optional<Record<ProcessInstanceRecordValue>> processInstanceRecord =
+        StreamFilter.processInstance(recordStreamSource.processInstanceRecords())
+            .withProcessInstanceKey(actual.getProcessInstanceKey())
+            .withBpmnElementType(BpmnElementType.PROCESS).stream()
+            .findFirst();
+
+    assertThat(processInstanceRecord.isPresent())
+        .withFailMessage("Process with key %s was not started", actual.getProcessInstanceKey())
+        .isTrue();
+
+    return this;
+  }
+
+  /**
+   * Verifies the expectation that the process instance is active.
+   *
+   * @return this {@link ProcessInstanceAssertions}
+   */
+  public ProcessInstanceAssertions isActive() {
+    final boolean isActive =
+        StreamFilter.processInstance(recordStreamSource.processInstanceRecords())
+            .withProcessInstanceKey(actual.getProcessInstanceKey())
+            .withBpmnElementType(BpmnElementType.PROCESS).stream()
+            .noneMatch(
+                record ->
+                    record.getIntent() == ProcessInstanceIntent.ELEMENT_COMPLETED
+                        || record.getIntent() == ProcessInstanceIntent.ELEMENT_TERMINATED);
+
+    assertThat(isActive)
+        .withFailMessage("Process with key %s is not active", actual.getProcessInstanceKey())
+        .isTrue();
+
+    return this;
+  }
+
+  /**
+   * Verifies the expectation that the process instance is completed.
+   *
+   * @return this {@link ProcessInstanceAssertions}
+   */
+  public ProcessInstanceAssertions isCompleted() {
+    final Optional<Record<ProcessInstanceRecordValue>> processInstanceRecord =
+        StreamFilter.processInstance(recordStreamSource.processInstanceRecords())
+            .withProcessInstanceKey(actual.getProcessInstanceKey())
+            .withBpmnElementType(BpmnElementType.PROCESS)
+            .withIntent(ProcessInstanceIntent.ELEMENT_COMPLETED).stream()
+            .findFirst();
+
+    assertThat(processInstanceRecord.isPresent())
+        .withFailMessage("Process with key %s was not completed", actual.getProcessInstanceKey())
+        .isTrue();
+
+    return this;
+  }
+
+  /**
+   * Verifies the expectation that the process instance is not completed.
+   *
+   * @return this {@link ProcessInstanceAssertions}
+   */
+  public ProcessInstanceAssertions isNotCompleted() {
+    final Optional<Record<ProcessInstanceRecordValue>> processInstanceRecord =
+        StreamFilter.processInstance(recordStreamSource.processInstanceRecords())
+            .withProcessInstanceKey(actual.getProcessInstanceKey())
+            .withBpmnElementType(BpmnElementType.PROCESS)
+            .withIntent(ProcessInstanceIntent.ELEMENT_COMPLETED).stream()
+            .findFirst();
+
+    assertThat(processInstanceRecord.isPresent())
```

Comment:
:x: 
This could be refactored to:
```
Stream<Record> completedElements = sharedCode();

a) actualCompleted = completedElements.findFirst().isPresent();
b) actualNotCompleted = completedEleemnts.findFirst().isEmpty();
```

Same for other methods below